invenio_rdm_migrator/state/base.py

```diff
@@ -15,24 +19,30 @@
 class State(ABC):
     """State interface."""
 
-    def __init__(self, filepath=None, validate=False):
+    def __init__(self, dbpath):
         """Constructor."""
-        self._data = {}
-
-        if filepath and filepath.exists():  # load state from file
-            start = ts(iso=False)
-            print(f"Loading state from {filepath} {ts()}")
-            with open(filepath, "r") as file:
-                self._data = json.loads(file.read())
-
-            print(f"Finished loading state {ts()}")
-            if validate:
-                print(f"Validating state {ts()}")
-                for _, entry in self._data.items():
-                    self._validate(entry)
-                print(f"Finished validating state {ts()}")
-            end = ts(iso=False)
-            print(f"State loading took {end-start} seconds.")
+
+        self.dbpath = dbpath
+
+        start = ts(iso=False)
+        print(f"Loading state from {dbpath} {ts()}")
+        # load file into memory
+        disk_file = create_engine(f"sqlite:///{dbpath}").connect()
+        # TODO: evaluate using ATTACH to avoid loading to a temporary file
```

Comment:
self note:
- https://www.sqlite.org/lang_attach.html
- https://stackoverflow.com/questions/62333314/python-sqlalchemy-in-memory-database-connect
- https://stackoverflow.com/questions/23036751/can-sqlalchemy-work-well-with-multiple-attached-sqlite-database-files
- https://docs.python.org/3/library/sqlite3.html
- https://docs.sqlalchemy.org/en/20/dialects/sqlite.html

Reply:
attach allows you to have two databases in the same connection, however, you still need to copy them. Therefore, is better to have them separated.